Foundation/GTMNSString+HTML.m

```diff
@@ -486,12 +486,14 @@ - (NSString *)gtm_stringByUnescapingFromHTML {
           NSScanner *scanner = [NSScanner scannerWithString:hexSequence];
           unsigned value;
           if ([scanner scanHexInt:&value] &&
-              value < USHRT_MAX &&
+              value < INT_MAX &&
               value > 0
               && [scanner scanLocation] == length - 4) {
-            unichar uchar = (unichar)value;
-            NSString *charString = [NSString stringWithCharacters:&uchar length:1];
-            [finalString replaceCharactersInRange:escapeRange withString:charString];
+            value = NSSwapHostIntToLittle(value);
+            NSString *charString = [[NSString alloc] initWithBytes:&value length:sizeof(value) encoding:NSUTF32LittleEndianStringEncoding];
```

Comment:
So most of GTM is manual reference counting and the [cocoapod is set up that way](https://github.com/google/google-toolbox-for-mac/blob/master/GoogleToolboxForMac.podspec#L19); so this will likely leak on most folks.  So you'll need a `[charString release]` below (since there isn't a `+stringWithBytes:length:encoding:`).

The other option would be to manually decompose it into surrogate pairs to use UTF-16 (`unichar`).

Reply:
Changed to calculate UTF-16 surrogate pairs (and use `+stringWithCharacters: length:`) for values from 0x10000 to 0x10FFFF.